cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -569,10 +572,20 @@ func StartControllers(ctx ControllerContext, startSATokenController InitFunc, co
 			klog.Warningf("Skipping %q", controllerName)
 			continue
 		}
-		if debugHandler != nil && unsecuredMux != nil {
-			basePath := "/debug/controllers/" + controllerName
-			unsecuredMux.UnlistedHandle(basePath, http.StripPrefix(basePath, debugHandler))
-			unsecuredMux.UnlistedHandlePrefix(basePath+"/", http.StripPrefix(basePath, debugHandler))
+		if ctrl != nil {
+			// To remain compatible with existing controllers, a nil value is allowed,
+			// and will be consider a anonymous controller that requests no additional
+			// features from the controller manager.
```

Comment:
Move this information into the godoc of initFn? It feels a bit out of place here at the callsite.

Reply:
Good points. Thanks! updated.